src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,202 @@
+
+using namespace std::string_view_literals;
+
+namespace AppInstaller::Repository::Rest::Schema::V1_0
+{
+    // Endpoint constants
+    constexpr std::string_view ManifestSearchPostEndpoint = "/api/manifestSearch?"sv;
+    constexpr std::string_view ManifestByVersionAndChannelGetEndpoint = "/api/packageManifests/"sv;
+
+    // General API response constants
+    constexpr std::string_view Data = "data"sv;
+
+    // Search body constants
+    constexpr std::string_view FetchAllManifests = "fetchAllManifests"sv;
+
+    // Search response constants
+    constexpr std::string_view PackageIdentifier = "PackageIdentifier"sv;
+    constexpr std::string_view PackageName = "PackageName"sv;
+    constexpr std::string_view Publisher = "Publisher"sv;
+    constexpr std::string_view PackageFamilyName = "PackageFamilyName"sv;
+    constexpr std::string_view ProductCode = "ProductCode"sv;
+    constexpr std::string_view Versions = "Versions"sv;
+    constexpr std::string_view Version = "version"sv;
+    constexpr std::string_view Channel = "Channel"sv;
+
+    namespace
+    {
+        utility::string_t GetJsonKeyNameString(std::string_view nodeName)
+        {
+            return utility::conversions::to_string_t(nodeName.data());
+        }
+
+        web::json::value GetSearchBody(const SearchRequest& searchRequest)
+        {
+            // TODO: Use search request to construct search body.
+            UNREFERENCED_PARAMETER(searchRequest);
+
+            web::json::value json_body;
+            json_body[GetJsonKeyNameString(FetchAllManifests)] = web::json::value::string(L"true");
+
+            return json_body;
+        }
+
+        std::string GetRestAPIBaseUri(std::string restApiUri)
+        {
+            if (restApiUri.back() == '/')
+            {
+                restApiUri.pop_back();
+            }
+
+            return restApiUri;
+        }
+
+        utility::string_t GetSearchEndpoint(const std::string& restApiUri)
+        {
+            std::string fullSearchAPI = restApiUri;
+            return utility::conversions::to_string_t(fullSearchAPI.append(ManifestSearchPostEndpoint));
+        }
+
+        utility::string_t GetManifestByVersionEndpoint(
+            const std::string& restApiUri, const std::string& packageId, const std::string& version, const std::string& channel)
+        {
+            std::string versionEndpoint = restApiUri;
+            versionEndpoint.append(ManifestByVersionAndChannelGetEndpoint).append(packageId);
+
+            // Add Version Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("?version=").append(version);
+            }
+
+            // Add Channel Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("&channel=").append(channel);
+            }
+
+            return utility::conversions::to_string_t(versionEndpoint);
+        }
+
+        bool IsStringWhitespace(const std::string& value)

```

Comment:
>bool IsStringWhitespace(const std::string& value) [](start = 8, length = 49)

I think we have this utility in AppInstallerStrings.hpp. If not, it's a good idea to move the utility there

Reply:
My bad, I forgot to delete this. I am already using Utiltity::IsStringEmptyOrWhitespace 

---
In reply to: [594712782](https://github.com/microsoft/winget-cli/pull/765#discussion_r594712782) [](ancestors = 594712782)

Reply:
Ah, I saw you already  used Utility::IsEmptyOrWhitespace(result) below. Then we can delete this 

---
In reply to: [594712782](https://github.com/microsoft/winget-cli/pull/765#discussion_r594712782) [](ancestors = 594712782)